### PR TITLE
Revert NMO handling in main

### PR DESCRIPTION
**Pull request by orenc1**
Upgrade is being blocked from previous version if NMO CRs exist on the cluster.
In main, we just remove the NMO CRD as a leftover using update patch mechanism.

Signed-off-by: orenc1 <ocohen@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
Revert NMO handling in main
```

### CODE REVIEW

File: controllers/hyperconverged/hyperconverged_controller.go [hunk 1]
```diff
@@ -481,41 +488,45 @@ func (r *ReconcileHyperConverged) handleUpgrade(req *common.HcoRequest, init boo
...
+				Status:             metav1.ConditionFalse,
+				Reason:             nmoCrExistErrorReason,
+				Message:            nmoCrExistErrorMessage,
+				ObservedGeneration: req.Instance.Generation,
+			})
+			r.eventEmitter.EmitEvent(req.Instance, corev1.EventTypeWarning, "NotUpgradeable",
+				fmt.Sprintf("Node Maintenance Operator custom resources %s have been found. Please remove them to allow upgrade. "+
+					"You can use NMO standalone operator if keeping the node(s) under maintenance is still required.", nmoCrdName))
+			req.Upgradeable = false
+			err = r.setOperatorUpgradeableStatus(req)
```
Comment (nunnatsa) [inline controllers/hyperconverged/hyperconverged_controller.go:525]: Not sure it should be called from here. The `Reconcile` method should handle this call.

**Comment by orenc1** [comment]
This PR should be done only on release-1.6, to block upgrade to 1.7.0 if NMO CR exists.
I'm opening a new one and modifying this one to remove NMO logic in main.

**Comment by orenc1** [comment]
hco-e2e-image-index-sno-aws lane passed.
/override ci/prow/hco-e2e-image-index-sno-azure
okd-hco-e2e-upgrade-index-aws lane passed.
/override ci/prow/okd-hco-e2e-upgrade-index-gcp

**Review by tiraboschi — APPROVED** [review]

**Review by tiraboschi — APPROVED** [review]

**Pull request merged**